### DISCUSSION TITLE
Problem using half cell model combined with particle mechanics submodel

### DISCUSSION THREAD

**Discussion by KAschad** [pybamm-team/PyBaMM#1924] [Q&A] [answered]
Hello PyBaMM-Team,

I am currently working on a battery cell model, that should represent the following cell: lithium metal anode / solid electrolyte / LFP (or LMO) cathode. The focus of my model is on the intercalation and thermally induced thickness change of the cathode. I have tried to build the DFN model as a half-cell model with the necessary submodels :

model = pybamm.models.full_battery_models.lithium_ion.DFN(options= {
        "working electrode": "positiv",
        "particle": "Fickian diffusion",
        "cell geometry": "arbitrary",
        "thermal": "lumped",
        "particle mechanics": "swelling only",
    })

Unfortunately, I get the following error when running the program: 
"Functions can only be applied to variables on the same domain"

Is it even possible to run the lithium metal model with the submodels listed above?

I am new to using PyBaMM as well as Python, therefore I want to apologize in advance, if the solution is completely obvious. Thank you very much in advance for your help.

Kind regards
Kathrin

**Comment by rtimms** [comment, marked as answer]
Hi turns out this is an issue with the thermal models and half cells. I've opened an issue #1927 and will look into fixing this. Thanks for pointing it out!

**Reply by rtimms** [reply]
I've opened a PR #1929 that fixes this

**Reply by KAschad** [reply]
Thanks a lot, now it works :)
